File: WebContent/WEB-INF/jython/action_reply_query.py
from cn.edustar.jitar.data import BaseQuery
from cn.edustar.jitar.util import ParamUtil

class ActionReplyQuery(BaseQuery):
    def __init__(self, selectFields):
        BaseQuery.__init__(self, selectFields)
        
        self.orderType = 0
        self.params = ParamUtil(request)
        self.actionId = None
        self.userId = None
        
    def initFromEntities(self, qctx):
        qctx.addEntity("ActionReply", "actr", "")
        
    def resolveEntity(self, qctx, entity):
        if "u" == entity:
            qctx.addEntity("User", "u", "actr.userId = u.userId")       
        else :
            BaseQuery.resolveEntity(self, qctx, entity);
            
    def applyWhereCondition(self, qctx):
        if (self.actionId != None):
            qctx.addAndWhere("actr.actionId = :actionId")
            qctx.setInteger("actionId", self.actionId)
        if (self.userId != None):
            qctx.addAndWhere("actr.userId = :userId")
            qctx.setInteger("userId", self.userId)
        
    def applyOrderCondition(self, qctx):
        if self.orderType == 0:
            qctx.addOrder("actr.actionReplyId ASC")

File: src/cn/edustar/jitar/service/ActionReplyQuery.java
package cn.edustar.jitar.service;

import cn.edustar.jitar.data.BaseQuery;
import cn.edustar.jitar.data.QueryContext;

public class ActionReplyQuery extends BaseQuery {

    public Integer orderType = 0;
    public Integer actionId = null;
    public Integer userId = null;

    public ActionReplyQuery(String selectFields) {
        super(selectFields);
    }

    @Override
    public void initFromEntities(QueryContext qctx) {
        qctx.addEntity("ActionReply", "actr", "");

    }

    public void resolveEntity(QueryContext qctx, String entity) {
        if ("u".equals(entity)) {
            qctx.addEntity("User", "u", "actr.userId = u.userId");
        } else {
            super.resolveEntity(qctx, entity);
        }
    }

    public void applyWhereCondition(QueryContext qctx) {
        if (this.actionId != null) {
            qctx.addAndWhere("actr.actionId = :actionId");
            qctx.setInteger("actionId", this.actionId);
        }
        if (this.userId != null) {
            qctx.addAndWhere("actr.userId = :userId");
            qctx.setInteger("userId", this.userId);
        }
    }
    public void applyOrderCondition(QueryContext qctx) {
        if (this.orderType == 0) {
            qctx.addOrder("actr.actionReplyId ASC");
        }
    }

}
